File: src/frontend/src/apis/services/audit-statistics.service.ts
import { apiClient } from '../client'
import {
  ResponseMessage,
  TrendPointDTO,
  EndpointRankingDTO,
  EndpointLatencyDTO,
} from '../schema/type'

export const auditStatisticsService = {
  async getTrends(period: string = '7d'): Promise<ResponseMessage<TrendPointDTO[]>> {
    const response = await apiClient.get<ResponseMessage<TrendPointDTO[]>>(
      '/admin/audit/statistics/trends',
      { params: { period } }
    )
    return response.data
  },

  async getEndpointRanking(
    period: string = '7d',
    limit: number = 20
  ): Promise<ResponseMessage<EndpointRankingDTO[]>> {
    const response = await apiClient.get<ResponseMessage<EndpointRankingDTO[]>>(
      '/admin/audit/statistics/endpoints',
      { params: { period, limit } }
    )
    return response.data
  },

  async getEndpointLatencyRanking(
    period: string = '7d',
    limit: number = 20
  ): Promise<ResponseMessage<EndpointLatencyDTO[]>> {
    const response = await apiClient.get<ResponseMessage<EndpointLatencyDTO[]>>(
      '/admin/audit/statistics/latency',
      { params: { period, limit } }
    )
    return response.data
  },
}

File: src/backend/src/main/java/com/bluenet/web/application/service/AuditStatisticsService.java
package com.bluenet.web.application.service;

import com.bluenet.web.api.dto.audit.EndpointLatencyDTO;
import com.bluenet.web.api.dto.audit.EndpointRankingDTO;
import com.bluenet.web.api.dto.audit.StatisticsPeriod;
import com.bluenet.web.api.dto.audit.TrendPointDTO;

import java.util.List;

/**
 * 审计统计服务接口，提供基于审计数据的聚合统计查询。
 * <p>
 * 当前实现直接查询 PostgreSQL，未来可通过预聚合方案替换实现。
 * </p>
 */
public interface AuditStatisticsService {
    List<TrendPointDTO> getTrends(StatisticsPeriod period);

    List<EndpointRankingDTO> getEndpointRanking(StatisticsPeriod period, int limit);

    List<EndpointLatencyDTO> getEndpointLatencyRanking(StatisticsPeriod period, int limit);
}
